nengo_spa/algebras/tvtb_algebra.py

```diff
@@ -124,7 +124,12 @@ def create_vector(self, d, properties, *, rng=None):
             properties.remove(TvtbProperties.UNITARY)
             v = self.make_unitary(rng.randn(d))
         elif TvtbProperties.POSITIVE in properties:
-            from scipy.linalg import sqrtm
+            try:
+                from scipy.linalg import sqrtm
+            except ImportError:
+                raise ValueError(
```

Comment:
Not sure that `ValueError` is the right exception type to use here ...
(But definitely nothing wrong with providing the more precise error message.)

Reply:
Right I was doing this to mirror https://github.com/nengo/nengo-spa/pull/271/files#diff-db444a08d43b9a69e4a19456a46ae35748e19b5e058b0e12a2955abda81eb3d3R225 which is also triggered by not having SciPy installed. I can see the reason for them being different, since the other situation is like a value error inside of an import error. I could also see them both being `ImportError` if you think of the import happening because it's fractional (even though the code has it the other way around).

Reply:
The analogy would be the value is the "positive" property in this case versus the fractional exponent value in the other case. Both exceptions require a particular value combined with SciPy being missing.

Reply:
I see, the reference is definitely relevant and needs to be considered together with this.

Reply:
I decided to use `ImportError` (or `ModuleNotFoundError` for Python >=3.6) in both places. While one could argue that the value is invalid given an environment without SciPy, it still comes down to not being able to import SciPy and if SciPy were available, the value would be valid. So the fix is not to fix the value, but provide SciPy.